shared/profile/edit-avatar/index.native.js

```diff
@@ -1,33 +1,127 @@
-        </ButtonBar>
-      </Box>
-    </StandardScreen>
-  )
-}
+const AVATAR_SIZE = 250
+
+class EditAvatar extends React.Component<Props> {
+  _h: number = 0
+  _w: number = 0
+  _x: number = 0
+  _y: number = 0
+
+  _onSave = () => {
+    const filename = isIOS ? this.props.image.uri.replace('file://', '') : this.props.image.path
+    if (this._h && this._w && this._x && this._y) {
+      this.props.onSave(filename, this._getCropCoordinates())
+      return
+    }
+    this.props.onSave(filename)
+  }
+
+  _getCropCoordinates = () => {
+    const x = this._x
+    const y = this._y
+    const rH = this.props.image.height / this._h
+    const rW = this.props.image.width / this._w
+    const x0 = rW * x
+    const y0 = rH * y
+    return {
+      x0: Math.round(x0),
+      x1: Math.round((x + AVATAR_SIZE) * rW),
+      y0: Math.round(y0),
+      y1: Math.round((y + AVATAR_SIZE) * rH),
+    }
+  }
 
-const styleCaption = {
-  marginTop: globalMargins.medium,
-  marginLeft: globalMargins.small,
-  marginRight: globalMargins.small,
-  textAlign: 'center',
+  _onZoom = ({height, width, x, y}: {height: number, width: number, x: number, y: number}) => {
+    this._h = height
+    this._w = width
+    this._x = x
+    this._y = y
+  }
+
+  render() {
+    return (
+      <StandardScreen
+        onCancel={this.props.onClose}
+        scrollEnabled={false}
+        style={container}
+        title="Zoom and pan"
+      >
+        <Box
+          style={{
+            marginBottom: globalMargins.small,
+            marginTop: globalMargins.small,
+          }}
+        >
+          <Box style={isIOS ? null : styles.zoomContainer}>
+            <ZoomableBox
+              bounces={false}
+              contentContainerStyle={{
+                height: this.props.image.height,
+                width: this.props.image.width,
+              }}
+              // Temporarily deactive zooming on Android.
+              maxZoom={isIOS ? 10 : 1}
+              onZoom={isIOS ? this._onZoom : null}
+              showsHorizontalScrollIndicator={false}
+              showsVerticalScrollIndicator={false}
+              style={styles.zoomContainer}
+            >
+              <NativeImage
+                resizeMode="contain"
+                source={{uri: `data:image/jpeg;base64,${this.props.image.data}`}}
+                style={{
+                  height: isIOS ? this.props.image.height : AVATAR_SIZE,
+                  width: isIOS ? this.props.image.width : AVATAR_SIZE,
+                }}
+              />
+            </ZoomableBox>
+          </Box>
+          <ButtonBar direction="column">
+            <WaitingButton
+              fullWidth={true}
+              label="Save"
+              onClick={this._onSave}
+              style={styles.button}
+              type="Primary"
+              waitingKey={null}
+            />
+          </ButtonBar>
+        </Box>
+      </StandardScreen>
+    )
+  }
 }
 
+const container = platformStyles({
```

Comment:
container should get in styleSheetCreate

Reply:
It's because `StandardScreen` is expecting an `Object` and not `StylesCrossPlatform`. Should I fix that as part of this PR?

I get this when I try to use it in `styleSheetCreate`:

```
Cannot create `StandardScreen` element because read-only array type [1] is incompatible with object type [2] in property `style`.
```

Reply:
short term i'd just change standard screen to take any instead of object. any loses typing but object is just incorrect

Reply:
I tried `any` and `?Object | ?StylesCrossPlatform`, and both end up with this error:

<img width="320" alt="screen shot 2018-06-29 at 3 15 01 pm" src="https://user-images.githubusercontent.com/30595/42110519-5d5dc31c-7baf-11e8-8f0c-5962e855fe94.png">

...which I don't understand. It's where the wrapped component's props are being expanded: https://github.com/keybase/client/blob/master/shared/common-adapters/header-hoc.native.js#L42

Reply:
thats not a flow error, thats an actual runtime error
in RN you can't do 
```
{
  foo: 'bar',
  ...(test ? {baz: whut} : null)
}
```
it doesn't allow you to spread null
also if your spreading the output of stylesheetcreate that won't work as its numbers and not an object. thats what collapseStyles is for

Reply:
Thanks for the explanation!